Source/objects.cpp

```diff
@@ -4897,7 +4873,7 @@ void OperateObject(int pnum, int i, bool teleFlag)
 		break;
 	case OBJ_SARC:
 	case OBJ_L5SARC:
-		OperateSarc(pnum, i, sendmsg);
+		OperateSarc(i, sendmsg, sendmsg);

```

Comment:
This really confused me until I saw SyncOpObject 😂

Reply:
Yeah, I wish I could have made it nicer, but assuming that they where the same was one of the reasons https://github.com/diasurgical/devilutionX/pull/4571 ended up being buggy. So need to test and understand things here a bit better and then maybe do a follow up. Still I think this is one step in the right direction.

Reply:
At the very least there is a decent chance that SyncOpObject and OperateObject can be combined instead use a couple of sensibly named variables. Kinda funny that the original had DeltaSync and SyncOpObject combined, but the simpler logic is the other way around :)